### DISCUSSION TITLE
Option to generate alternate links in <head>

### DISCUSSION THREAD

**Discussion by beeb** [ivanhofer/typesafe-i18n#547] [DX improvements]
Is there a way that we could generate alternative links for the same page in other languages inside the `<head>` tag? 

Reference: https://developers.google.com/search/docs/specialty/international/localized-versions#html

`<link rel="alternate" hreflang="lang_code" href="url_of_page" />`

I've seen i18n libraries for Astro provide a component for this, but I don't know how it would look with `typesafe-i18n`. Specifically, I would be interested in a solution for SvelteKit.

Thanks!

**Comment by ivanhofer** [comment]
Hi @beeb,
there is currently no opinionated solution for this. Please take a look [here](https://kit.svelte.dev/docs/seo#manual-setup-title-and-meta).
You can use `svelte:head` to put your alternate links into the document. You can probably wrap the functionality into a component with a few lines of code.
Fell free to share your solution here :)

**Reply by beeb** [reply]
Thanks for this! I managed to do something like you suggested. I'll make a PR to the sveltekit demo repo tomorrow including this and an example how to change the <title> element.

`HeadHrefLangs.svelte`

```svelte
<script lang="ts">
	import { locales } from '$i18n/i18n-util'
	import { page } from '$app/stores'
	import { replaceLocaleInUrl } from '$lib/utils'
</script>

{#each locales as l}
	<link rel="alternate" hreflang={l} href={`${replaceLocaleInUrl($page.url, l)}`} />
{/each}
```

`+layout.svelte`

```svelte
<script lang="ts">
import HeadHrefLangs from '$lib/components/HeadHrefLangs.svelte'
</script>
<svelte:head>
	<HeadHrefLangs />
</svelte:head>
```

`lib/utils.ts`

```ts
export const replaceLocaleInUrl = (url: URL, locale: string): string => {
	const [, , ...rest] = url.pathname.split('/')
	return `/${[locale, ...rest].join('/')}${url.search}`
}
```
